### ISSUE TITLE
Ratel reports errors on predicate delete

### ISSUE THREAD

**Issue by catroot** [dgraph-io/dgraph#2081]
My setup is three server node (different machines) and one zero node.
On single node setup this error absent.
![graph screenshot_1](https://user-images.githubusercontent.com/10930458/35729154-dcb5a7d8-081e-11e8-9180-5f1394b6bf53.jpg)

Tried to load dataset in to cluster:
```
#dgraph live -r 21million.rdf.gz --zero 192.168.8.172:3080

Processing 21million.rdf.gz
Total Txns done:        0 RDFs per second:    9998 Time Elapsed: 2s, Aborts: 0
Total Txns done:        0 RDFs per second:    7500 Time Elapsed: 4s, Aborts: 0
2018/02/02 13:17:03 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:03 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
Total Txns done:        0 RDFs per second:    6666 Time Elapsed: 6s, Aborts: 2
2018/02/02 13:17:03 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:04 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:04 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:04 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:04 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:04 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:05 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:05 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:05 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
Total Txns done:        0 RDFs per second:    7500 Time Elapsed: 8s, Aborts: 11
2018/02/02 13:17:06 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:06 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:06 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:06 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:06 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:07 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:07 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
Total Txns done:        0 RDFs per second:    8000 Time Elapsed: 10s, Aborts: 18
2018/02/02 13:17:07 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:07 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:07 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:08 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:09 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
Total Txns done:        1 RDFs per second:    7500 Time Elapsed: 12s, Aborts: 29
2018/02/02 13:17:10 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:10 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:10 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:10 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:10 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:11 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:11 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
Total Txns done:        2 RDFs per second:    7142 Time Elapsed: 14s, Aborts: 36
2018/02/02 13:17:11 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:11 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:11 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:12 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:12 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:12 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:12 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
2018/02/02 13:17:12 batch.go:133: Error while mutating rpc error: code = Unknown desc = Conflicts with pending transaction. Please abort.
^C
```

**Comment by pawanrawal** [comment]
@catroot It would be great if you could share the steps to reproduce this issue.

Were you updating the `friend` edge by removing reverse/count and then encountered the error about `Could not delete predicate`? Did you try loading the `21million.rdf.gz` after you got this error or before?

**Comment by catroot** [comment]
Just after join two another nodes to the old one with some data on it.
After cleaning up database, loading a few data from 21million dataset and try again - the same issue.

**Comment by pawanrawal** [comment]
Can you share the commands that you used to run Dgraph and Zero? Did you have any replicas?

**Comment by catroot** [comment]
No replicas. 
Dgraph zero startup:
dgraph zero --port_offset -2000 --my 192.168.8.172:3080

Dgraph server startups:
dgraph server --memory_mb 1048 --zero 192.168.8.172:3080 --my 192.168.8.172:7080
dgraph server --memory_mb 1048 --zero 192.168.8.172:3080 --my 192.168.8.168:7080
dgraph server --memory_mb 1048 --zero 192.168.8.172:3080 --my 192.168.8.174:7080

**Comment by pawanrawal** [comment]
I was able to reproduce the error. This happens after a predicate move when the server receiving the drop attr request is not the one serving the predicate.

**Comment by pawanrawal** [comment]
I have a fix in https://github.com/dgraph-io/dgraph/pull/2114 which should be merged by tomorrow.

**Comment by pawanrawal** [comment]
I have pushed a fix in 77ad97efeea4ae0d4a15b3758e51cd2ee938c2da. It would be part of the nightly binary and also the next release.

**Comment by catroot** [comment]
Thank you.
